src/db/tables/attributions.ts

```diff
@@ -16,6 +16,28 @@ async function getAttributionsForSubscriber(subscriberId: number) {
     .returning("*")) as AttributionRow[];
 }
 
+async function getLatestAttributionForSubscriberWithType(
+  subscriberId: number,
+  type: string,
+) {
+  logger.info("getLatestAttributionForSubscriberWithType", {
+    subscriberId,
+    type,
+  });
+  const res = await knex("attributions")
+    .orderBy("created_at")
+    .where("subscriber_id", subscriberId)
+    .andWhere("type", type)
+    .returning([
+      "entrypoint",
+      "utm_source",
+      "utm_medium",
+      "utm_term",
+      "utm_campaign",
+    ]);
+  return res[0] as Partial<AttributionRow>;
```

Comment:
If you add the following to the `Tables` interface in `knex-tables.d.ts`, then you don't need the type assertion - TypeScript can automatically derive the correct return value:

```ts

    attributions: Knex.CompositeTableType<
      AttributionRow,
      // On updates, auto-generated columns cannot be set, and nullable columns are optional:
      Omit<
        AttributionRow,
        AttributionAutoInsertedColumns | AttributionOptionalColumns
      > &
        Partial<Pick<AttributionRow, AttributionOptionalColumns>>,
      // On updates, don't allow updating the ID and created date; all other fields are optional, except updated_at:
      Partial<Omit<AttributionRow, "id" | "created_at">> &
        Pick<AttributionRow, "updated_at">
    >;
```

Reply:
Fixed thanks